webapp/tasks.py

```diff
@@ -1,53 +1,64 @@
 import time
 from multiprocessing import Process
+from typing import Dict
 
 import yaml
+from flask import Flask
 
+from webapp.models import Project, User, Webpage, db
 from webapp.site_repository import (
     SiteRepository,
     SiteRepositoryError,
     site_repository_exists,
 )
 
 # Keep track of generation tasks for each tree.
-TREE_TASKS = {}
+TREE_TASKS: Dict[str, Process] = {}
 
-def init_tasks(app, cache):
+def init_tasks(app: Flask):
     """
     Start event loop.
     """
-    Process(
-        target=load_site_trees,
-        args=(app, cache),
-    ).start()
 
+    @app.before_request
+    def start_tasks():
+        # Only run once
+        app.before_request_funcs[None].remove(start_tasks)
+        Process(
+            target=load_site_trees,
+            args=(app,),
+        ).start()
 
-def load_site_trees(app, cache):
+
+def load_site_trees(app: Flask):
     """
     Load the site trees from the queue.
     """
     while True:
         with open(app.config["BASE_DIR"] + "/" + "sites.yaml") as f:
             data = yaml.safe_load(f)
             for site in data["sites"]:
-                add_tree_task(site, "main", app, cache)
+                add_tree_task(site, "main", app)
```

Comment:
are we still going to return cached data for a certain amount of time instead of querying DB on every request?

Reply:
Yes, I just moved the cache object to the app config instead of passing it directly